packages/flame/test/components/mixins/has_paint_test.dart

```diff
@@ -0,0 +1,207 @@
+import 'dart:ui';
+
+import 'package:flame/components.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+class MyComponent extends PositionComponent with HasPaint {}
```

Comment:
I think it would we could make the tests clearer by having two components, one with HasPaint and a single paint and one with `HasPaint<String>` and type safe (or even an enum).